src/acpi.rs

```diff
@@ -419,3 +424,357 @@ pub fn process_msct() -> (
+    }
+}
+
+/// Convert u64 to MappingAttributes.
+impl From<u64> for MappingAttributes {
+    fn from(mode: u64) -> MappingAttributes {
+        MappingAttributes::from_bits_truncate(mode)
+    }
+}
+
+/// Convert MappingAttributes to u64.
+impl From<MappingAttributes> for u64 {
+    fn from(mode: MappingAttributes) -> u64 {
+        mode.bits()
+    }
+}
+
+struct Guid {
+    data1: u32,
+    data2: u16,
+    data3: u16,
+    index8: u8,
+    index9: u8,
+    index10: u8,
+    index11: u8,
+    index12: u8,
+    index13: u8,
+    index14: u8,
+    index15: u8,
+}
+
+impl fmt::Debug for Guid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{{ {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X} }}",
+            self.data1,
+            self.data2,
+            self.data3,
+            self.index8,
+            self.index9,
+            self.index10,
+            self.index11,
+            self.index12,
+            self.index13,
+            self.index14,
+            self.index15
+        )
+    }
+}
+
+fn from_slice_u8_to_Guid(slice: &[u8]) -> Guid {
+    let p: *const [u8; core::mem::size_of::<Guid>()] =
+        slice.as_ptr() as *const [u8; core::mem::size_of::<Guid>()];
+    unsafe { core::mem::transmute(*p) }
+}
+
+// To parse DeviceHandle in ACPI_NFIT_MEMORY_MAP and ACPI_NFIT_FLUSH_ADDRESS subtables
+const ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK: u32 = 0xf;
+const ACPI_NFIT_DEVICE_HANDLE_12BIT_MASK: u32 = 0xfff;
+
+// NFIT subtable type = 0x0
+fn parse_nfit_spa_range_structure(entry: *const ACPI_NFIT_SYSTEM_ADDRESS) -> MemoryDescriptor {
+    let mut mem_desc = MemoryDescriptor::default();
+    unsafe {
+        assert_eq!((*entry).Header.Type, 0);
+        debug!(
+            "ACPI_NFIT_SYSTEM_ADDRESS - Type {}, Length {},
+                RangeIndex: {},
+                Flags: {},
+                ProximityDomain: {},
+                RangeGuid: {:x?},
+                Address: 0x{:x},
+                Length: {} Bytes,
+                MemoryMapping: {:?}",
+            (*entry).Header.Type,
+            (*entry).Header.Length,
+            (*entry).RangeIndex,
+            (*entry).Flags,
+            (*entry).ProximityDomain,
+            from_slice_u8_to_Guid(&(*entry).RangeGuid),
+            (*entry).Address,
+            (*entry).Length,
+            MappingAttributes::from((*entry).MemoryMapping)
+        );
+
+        assert_eq!(
+            (*entry).Length % BASE_PAGE_SIZE as u64,
+            0,
+            "Not multiple of page-size."
+        );
+        mem_desc.ty = MemoryType::PERSISTENT_MEMORY;
+        mem_desc.phys_start = (*entry).Address;
+        mem_desc.page_count = (*entry).Length / BASE_PAGE_SIZE as u64;
+        mem_desc.att = MemoryAttribute::from_bits_truncate((*entry).MemoryMapping);
+
+        mem_desc
+    }
+}
+
+// NFIT subtable type = 0x1
+fn parse_nfit_region_mapping_structure(entry: *const ACPI_NFIT_MEMORY_MAP) {
```

Comment:
Doesn't parse anything? Maybe rename to log_nfit_region_mapping_structure?

Reply:
Well, eventually, we will use these functions, but I get it.

Updated.